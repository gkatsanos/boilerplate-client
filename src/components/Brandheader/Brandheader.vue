<template>
  <v-toolbar app
             scroll-off-screen
             fixed
             dark
             class="primary">
    <v-toolbar-title>Missed Connections</v-toolbar-title>
    <v-spacer></v-spacer>
    {{ user.firstName }}
    <v-menu offset-y v-if="isAuthenticated">
      <v-btn icon slot="activator" dark>
        <v-icon large dark>account_circle</v-icon>
      </v-btn>
      <v-list>
        <v-list-tile avatar>
          <v-list-tile-avatar>
            <v-icon medium>account_circle</v-icon>
          </v-list-tile-avatar>
          <v-list-tile-content>
            <v-list-tile-title>{{ user.firstName }} {{ user.lastName }}</v-list-tile-title>
            <v-list-tile-sub-title>{{ user.email }}</v-list-tile-sub-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
      <v-list>
        <v-list-tile>
          <v-btn block
                 primary
                 small
                 @click="submitLogout()">Logout
          </v-btn>
        </v-list-tile>
      </v-list>
    </v-menu>
    <signup v-if="!isAuthenticated"></signup>
    <login v-if="!isAuthenticated"></login>
  </v-toolbar>
</template>

<script>
  import { mapGetters, mapState } from 'vuex';
  import signup from '../Signup/Signup';
  import login from '../Login/Login';

  export default {
    name: 'brandheader',
    components: {
      signup,
      login,
    },
    computed: {
      ...mapGetters([
        'isAuthenticated',
      ]),
      ...mapState([
        'user',
      ]),
    },
    methods: {
      submitLogout() {
        this.$store.dispatch('logout');
      },
    },
  };
</script>
