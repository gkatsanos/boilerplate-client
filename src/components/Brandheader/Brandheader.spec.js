import Vuex from 'vuex';
import Vuetify from 'vuetify';
import { mount, createLocalVue } from 'vue-test-utils';
import Brandheader from '@/components/Brandheader/Brandheader';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(Vuetify);


describe('Brandheader', () => {
  let cmp;
  let template;
  let store;
  let state;
  let actions;
  let getters;

  beforeEach(() => {
    actions = {};
    getters = {
      isAuthenticated: () => true,
    };
    state = {
      user: {
        firstName: 'Georgios',
        lastName: 'Katsanos',
        email: 'gkatsanos@gmail.com',
      },
    };
    store = new Vuex.Store({
      state,
      actions,
      getters,
    });
    cmp = mount(Brandheader, {
      store,
      localVue,
    });
  });

  it('has the expected html structure', () => {
    template = cmp.html();
    expect(template).toMatchSnapshot();
  });
});
