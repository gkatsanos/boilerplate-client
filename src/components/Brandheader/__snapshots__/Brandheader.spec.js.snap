// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brandheader has the expected html structure 1`] = `
<nav class="primary toolbar toolbar--fixed theme--dark" style="margin-top: 0px; padding-left: 0px; padding-right: 0px;">
    <div class="toolbar__content" style="height: 56px;">
        <div class="toolbar__title">Missed Connections</div>
        <div class="spacer"></div>
        Georgios
        <div class="menu" style="display: inline-block;">
            <div class="menu__activator"><button type="button" class="btn btn--icon btn--raised theme--dark" data-ripple="true">
                    <div class="btn__content"><i class="material-icons icon icon--large theme--dark">account_circle</i></div>
                </button></div>
            <div class="menu__content" style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;">
                <ul data-uid="7" class="list">
                    <li>
                        <div class="list__tile list__tile--avatar">
                            <div class="avatar list__tile__avatar" style="height: 48px; width: 48px;"><i class="material-icons icon icon--medium">account_circle</i></div>
                            <div class="list__tile__content">
                                <div class="list__tile__title">Georgios Katsanos</div>
                                <div class="list__tile__sub-title">gkatsanos@gmail.com</div>
                            </div>
                        </div>
                    </li>
                </ul>
                <ul data-uid="9" class="list">
                    <li>
                        <div class="list__tile"><button type="button" class="btn btn--block btn--raised btn--small primary" data-ripple="true">
                                <div class="btn__content">Logout
                                </div>
                            </button></div>
                    </li>
                </ul>
            </div>
        </div>
        <!---->
        <!---->
    </div>
</nav>
`;
